WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,352 @@
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4.0',
+			),
+		),
+		'wp_upload_bits' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '2.0.0',
+			),
+		),
+		'xfn_check' => array(
+			3 => array(
+				'value'   => '',
+				'version' => '2.5.0',
+			),
+		),
+
+	); // End $target_functions.
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$paramCount = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $parameter_args ) {
+
+			if ( $position > $paramCount ) {
+				break;
+			}
+
+			if ( ! isset( $parameters[ $position ] ) ) {
+				continue;
+			}
+
+			switch ( $parameters[ $position ]['raw'] ) {
+				case 'true':
+					$matched_parameter = true;
+					break;
+				case 'false':
+					$matched_parameter = false;
+					break;
+				case 'null':
+					$matched_parameter = null;
+					break;
+				case 'array()':
+					$matched_parameter = array();
+					break;
+				default:
+					$matched_parameter = $this->strip_quotes( $parameters[ $position ]['raw'] );
+					break;
+			}
+
+			if ( $parameter_args['value'] === $matched_parameter ) {
+				continue;
+			}
+
+			$is_dynamic_parameter = $this->phpcsFile->findNext(
+				array( T_CONSTANT_ENCAPSED_STRING, T_ARRAY, T_FALSE, T_TRUE, T_NULL, T_LNUMBER, T_WHITESPACE ),
+				$parameters[ $position ]['start'],
+				( $parameters[ $position ]['end'] + 1 ),
+				true,
+				null,
+				true
+			);
+
+			if ( $is_dynamic_parameter ) {
+				$message = 'The dynamic parameter [%s] at position #%s of %s() could possibly be deprecated.';
+				$is_error = false;
+				$code = $this->string_to_errorcode( ucfirst( $matched_content ) . 'PossiblyDeprecated' );
+			} else {
+				$message = 'The parameter [%s] at position #%s of %s() has been deprecated since WordPress version %s.';
+				$is_error = version_compare( $parameter_args['version'], $this->minimum_supported_version, '<' );
+				$code = $this->string_to_errorcode( ucfirst( $matched_content ) . 'Found' );
+			}
+
+			$data    = array(
+				$parameters[ $position ]['raw'],
+				$position,
+				$matched_content,
+				$parameter_args['version'],
+			);
+
+			if ( isset( $parameter_args['value'] ) && $position < $paramCount ) {
+				$message .= ' Use "%s" instead.';
+				$data[]   = (string) $parameter_args['value'];
+			} else {
+				$message .= ' Instead do not pass the parameter.';
+			}
+
+			$this->addMessage( $message, $stackPtr, $is_error, $code, $data, 0 );
+		} // End foreach().
```

Comment:
The `// End foreach()` can be removed, the requirement for it has been removed from the Core standard.